### PR TITLE
[build] Add a parameter to specify sonic version during build

### PR DESCRIPTION
**Pull request by stepanblyschak**
Introduced a new build parameter 'SONIC_IMAGE_VERSION' that allows build
system users to build SONiC image with a specific version string. If
'SONIC_IMAGE_VERSION' was not passed by the user, SONIC_IMAGE_VERSION will be
set to the output of functions.sh:sonic_get_version function.

Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>

**- Why I did it**

**- How I did it**

**- How to verify it**

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [x] 201911
- [x] 202006

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Review by volodymyrsamotiy — DISMISSED** [review]

File: slave.mk [hunk 1]
```diff
@@ -219,6 +223,7 @@ $(info "SONIC_PROFILING_ON"              : "$(SONIC_PROFILING_ON)")
 $(info "KERNEL_PROCURE_METHOD"           : "$(KERNEL_PROCURE_METHOD)")
 $(info "BUILD_TIMESTAMP"                 : "$(BUILD_TIMESTAMP)")
 $(info "BUILD_LOG_TIMESTAMP"             : "$(BUILD_LOG_TIMESTAMP)")
+$(info "SONIC_IMAGE_VERSION"                   : "$(SONIC_IMAGE_VERSION)")
```
Comment (jleveque) [inline slave.mk:226]: Please align the second column

Comment (stepanblyschak) [reply slave.mk]: Done

**Review by yxieca — APPROVED** [review]

**Comment by yxieca** [comment]
@stepanblyschak did you test build with and without specifying SONIC_IMAGE_VERSION and see expected behavior in both cases?

**Comment by liat-grozovik** [comment]
@jleveque with the indentation fix, is it approved?

**Review by jleveque — APPROVED** [review]

**Comment by stepanblyschak** [comment]
@yxieca Yes, double checked:

Without specifying SONIC_IMAGE_VERSION:
```
"CONFIGURED_ARCH"                 : "amd64"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "1"
"SONIC_CONFIG_MAKE_JOBS"          : "12"
"SONIC_USE_DOCKER_BUILDKIT"       : ""
"USERNAME"                        : "admin"
"PASSWORD"                        : "YourPaSsWoRd"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_ZTP"                      : ""
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : "20200916.081408"
"BUILD_LOG_TIMESTAMP"             : "none"
"SONIC_IMAGE_VERSION"             : "master.0-cae8173e"
"BLDENV"                          : "buster"
"VS_PREPARE_MEM"                  : "yes"
"INCLUDE_MGMT_FRAMEWORK"          : "y"
"INCLUDE_ICCPD"                   : "n"
"INCLUDE_SYSTEM_TELEMETRY"        : "y"
"INCLUDE_HOST_SERVICE"            : "n"
"INCLUDE_RESTAPI"                 : "n"
"INCLUDE_SFLOW"                   : "y"
"INCLUDE_NAT"                     : "y"
"INCLUDE_KUBERNETES"              : "n"
"TELEMETRY_WRITABLE"              : ""

fatal: not a git repository (or any of the parent directories): .git
"SONIC_DPKG_CACHE_METHOD"         : "none"
```

I checked publicly built image by this PR, the image version is generated as previously.

With specifying SONIC_IMAGE_VERSION:
```
$ make SONIC_BUILD_JOBS=12 SONIC_IMAGE_VERSION=stepan target/sonic-mellanox.bin
SONiC Build System

Build Configuration
"CONFIGURED_PLATFORM"             : "mellanox"
"CONFIGURED_ARCH"                 : "amd64"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "12"
"SONIC_CONFIG_MAKE_JOBS"          : "12"
"SONIC_USE_DOCKER_BUILDKIT"       : ""
"USERNAME"                        : "admin"
"PASSWORD"                        : "YourPaSsWoRd"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_ZTP"                      : ""
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : "20200916.081528"
"BUILD_LOG_TIMESTAMP"             : "none"
"SONIC_IMAGE_VERSION"             : "stepan"
"BLDENV"                          : "buster"
"VS_PREPARE_MEM"                  : "yes"
"INCLUDE_MGMT_FRAMEWORK"          : "y"
"INCLUDE_ICCPD"                   : "n"
"INCLUDE_SYSTEM_TELEMETRY"        : "y"
"INCLUDE_HOST_SERVICE"            : "n"
"INCLUDE_RESTAPI"                 : "n"
"INCLUDE_SFLOW"                   : "y"
"INCLUDE_NAT"                     : "y"
"INCLUDE_KUBERNETES"              : "n"
"TELEMETRY_WRITABLE"              : ""

fatal: not a git repository (or any of the parent directories): .git
"SONIC_DPKG_CACHE_METHOD"         : "none"

$  strings target/sonic-mellanox.bin | grep -m 1 '^image_version' | sed -n 's/^image_version=\"\(.*\)\"$/\1/p'
stepan


```

**Pull request merged**